File: looking/julia/derivative.jl
# See https://en.wikipedia.org/wiki/Derivative
# Examples from https://github.com/JuliaSymbolics/Symbolics.jl

using Symbolics

@variables x y z

expressions = [
    :(Symbolics.derivative(x + x^2, x))
    :(Symbolics.derivative(x + 2x^2 + 6x^3 - 5x^4, x))
    :(Symbolics.derivative(cos(x) * sin(x), x))
    :(Symbolics.derivative(exp(x), x))
    :(Symbolics.derivative(log(x), x))
    :(Symbolics.jacobian([x + x*y, x^4 + y], [x, y]))
    :(Symbolics.hessian(x^4 + z, [x, y, z]))
]

for expr in expressions
    println("[] :julia-result \"", escape_string(string(expr)), " = ", eval(expr), "\".")
end

File: looking/python/derivative.py
# See https://en.wikipedia.org/wiki/Derivative

from sympy import diff, cos, exp, log, hessian
from sympy.matrices import Matrix
from sympy.abc import x, y, z

if __name__ == "__main__":
    cases = [
        "diff(cos(x), x)",
        "diff(x**4, x, x, x)",
        "diff(exp(x*y*z), x, y, y, z, z, z, z)",
        "diff((x + 1)*((x**2 + 2)*(x**3 + 3)), x)",
        "diff(x/x/x/x/x/x/x/x/x/x, x)",
        "diff(log(log(log(log(log(log(log(log(log(log(x)))))))))), x)",
        "diff(x*x*x*x*x*x*x*x*x*x, x)",
        "Matrix([x**3 + 5*y**4 - 9*z, exp(x*y*z)]).jacobian([x, y, z])",
        "hessian(x**3 + 5*y**4 - 9*z, [x, y, z])"
    ]

    for c in cases:
        print('[] :python-result "%s = %s".' % (c, eval(c)))
